transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/AddTransferMethodFragment.java

```diff
@@ -188,6 +192,15 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         mCreateTransferMethodButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
+                Map<String, String> params = new HashMap<>();
+                params.put(InsightEventTag.InsightEventTagEventParams.TRANSFER_METHOD_COUNTRY, mCountry);
+                params.put(InsightEventTag.InsightEventTagEventParams.TRANSFER_METHOD_CURRENCY, mCurrency);
+                params.put(InsightEventTag.InsightEventTagEventParams.TRANSFER_METHOD_TYPE, mTransferMethodType);
+                params.put(InsightEventTag.InsightEventTagEventParams.TRANSFER_METHOD_PROFILE_TYPE,
+                        mTransferMethodProfileType);
```

Comment:
we can make this code cleaner if we do like this collection initialization
```
Map<String, String> params = new HashMap<String, String>() {{
           put("key", "value"); 
}};
```

Reply:
@peter-joseph yes we can do it.
But double brace initialization has some disadvantages in java. They are in reference links,
Mainly **it may cause memory leaks**.
Kindly confirm it.
Reference
1)https://www.java67.com/2016/01/how-to-initialize-hashmap-with-values-in-java.html
2)https://javarevisited.blogspot.com/2015/10/what-is-double-brace-initialization-in-java-example-anti-pattern.html
3)https://stackoverflow.com/questions/1958636/what-is-double-brace-initialization-in-java

Reply:
good catch 🌮 I agree let's change ⚙️ 
Any suggestion to make this cleaner? So the goal is just call insight once in the fragment, like one liner code in each fragment, I also did not like to add constants in fragment

Reply:
updated as per our discussion.